From: Ben Hutchings <ben@decadent.org.uk>
Date: Sat, 27 Aug 2011 15:09:53 +0100
Subject: [PATCH] Revert "net/ipv4: Check for mistakenly passed in non-IPv4
 address"

This reverts commit d0733d2e29b652b2e7b1438ececa732e4eed98eb,
cherry-picked in 2.6.32.43.  I suspect that some applications
accidentally depend on the old behaviour, and we don't want to break
them in a stable update.
---
 net/ipv4/af_inet.c |    3 ---
 1 files changed, 0 insertions(+), 3 deletions(-)

diff --git a/net/ipv4/af_inet.c b/net/ipv4/af_inet.c
index a289878..57737b8 100644
--- a/net/ipv4/af_inet.c
+++ b/net/ipv4/af_inet.c
@@ -460,9 +460,6 @@ int inet_bind(struct socket *sock, struct sockaddr *uaddr, int addr_len)
 	if (addr_len < sizeof(struct sockaddr_in))
 		goto out;
 
-	if (addr->sin_family != AF_INET)
-		goto out;
-
 	chk_addr_ret = inet_addr_type(sock_net(sk), addr->sin_addr.s_addr);
 
 	/* Not specified by any standard per-se, however it breaks too
-- 
1.7.5.4

